### ISSUE TITLE
quantize_block always re-runs collect_reference even when caller has FP16 outputs cached

### ISSUE THREAD

**Issue by xesdiny** [intel/auto-round#2288]
# intel/auto-round Issue: Add `reference_output` parameter to `quantize_block` to allow callers to skip the collect_reference forward pass

## Summary

`CompressionOrchestrator.quantize_block` always runs an internal FP16 forward pass
("collect_reference / Step 3") to obtain reference outputs for the SignSGD loss
computation. This is redundant when the caller already holds the identical tensors
from a prior pass. Exposing a `reference_output=None` parameter would allow callers
to skip the forward pass entirely, saving significant CPU RAM and wall-clock time for
large models.

---

## Context: Integration with LLM-Compressor's SequentialPipeline

[vllm-project/llm-compressor](https://github.com/vllm-project/llm-compressor) uses auto-round
via `AutoRoundModifier`, which calls `quantize_block` once per transformer block. The
`SequentialPipeline` runs layer-by-layer and caches each subgraph's FP16 output in an
`IntermediatesCache` to feed as input to the next subgraph.

When `propagate_error=False` (the auto-round default), the same unquantized inputs flow
to every subgraph. Therefore:

```
IntermediatesCache[k+1]  =  FP16 output of block k (unquantized inputs, unquantized weights)
collect_reference output =  FP16 output of block k (unquantized inputs, unquantized weights)
```

Same block + same input + same weights → **outputs are mathematically identical**.
`collect_reference` re-derives them from scratch, duplicating the memory.

---

## Measured Impact

For Qwen3.5-MoE 122B (hidden=7168, seq=6144, N=8192, 8-GPU DDP):

| Item | CPU RAM / rank |
|------|---------------:|
| IntermediatesCache (N/8 = 1024 samples) | ~39 GB |
| collect_reference duplicate | ~39 GB |
| **Total duplication** | **~39 GB/rank × 8 ranks = ~312 GB** |

With ~310 GB already consumed by the cache + model weights, the duplicate pushes the
node over its 1.8 TiB physical RAM limit → OS SIGKILL → NCCL watchdog timeout.

---

### `auto_round/compressors/orchestrator.py` — `CompressionOrchestrator.quantize_block`

Add `reference_output=None` parameter and pass it through to `compress_block`:

```python
def quantize_block(
    self,
    block,
    inputs,
    q_input=None,
    device="cpu",
    auto_offload=True,
    reference_output=None,   # NEW
) -> Any:
    """...
    reference_output: Optional pre-computed FP16 reference outputs (list of tensors,
        one per calibration sample). When provided, the internal collect_reference
        forward pass in Step 3 is skipped, saving significant peak CPU RAM on large
        models. Supplied by LLM-Compressor's SequentialPipeline via
        ``AutoRoundModifier.set_fp_ref_outputs()``.
    ..."""
    ...
    new_q_input, reference_output = self.alg_composer.compress_block(
        block, input_ids, input_others,
        block_ctx=ctx, q_inputs=q_input,
        reference_output=reference_output,   # NEW
    )
```

### `auto_round/algorithms/composer.py` — `AlgorithmComposer.compress_block`

Add `reference_output=None` parameter and skip the forward pass when provided:

```python
def compress_block(
    self,
    block,
    fp_inputs,
    input_others,
    block_ctx: BlockContext,
    q_inputs=None,
    input_ids=None,
    reference_output=None,   # NEW
    **kwargs,
) -> tuple:
    ...
    reference_next_input = None
    # Step 3: Quantizer calibration
    if fp_inputs is not None:
        with torch.no_grad():
            quant_hooks = self._get_fp_act_hooks(block)
            if reference_output is None:          # NEW: skip if pre-seeded
                reference_output = block_forward_fn(block, fp_inputs, input_others)
            reference_next_input = getattr(block_forward_fn, "last_output_dict", None) or reference_output
            for h in quant_hooks:
                h.remove()
```

---

## Backward Compatibility

- `reference_output=None` is the default; existing callers that don't pass it retain
  **identical behaviour** — the forward pass runs as before.
- The parameter is keyword-only in practice (passed as `reference_output=...`), so
  there is no positional-argument breakage.

---

## Validation

Tested end-to-end on Qwen3-8B (W4A16, nsamples=16, iters=5, single GPU) with
llm-compressor PR vllm-project/llm-compressor#3130 and the local auto-round patch:
- All 37 transformer blocks quantized successfully
- No `TypeError` for unexpected keyword argument
- No OOM; collect_reference forward pass skipped per block
- Saved model identical in structure to an unpatched baseline

---

## Related

- LLM-Compressor companion PR: vllm-project/llm-compressor#3130
- LLM-Compressor Issue: vllm-project/llm-compressor#3129
